### PR TITLE
Add changelog entry for character count i18n options

### PR DESCRIPTION
**Pull request by romaricpascal**
Adds missing entry in the changelog for the options letting users configure the character count messages. I tried to group things a little as there are many similar options.

~~Leaving it as draft to have a second read tomorrow morning before opening it to review.~~ It's ready to review.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
 
 This was added in [pull request #2742: Add ability to customise character count fallback text](https://github.com/alphagov/govuk-frontend/pull/2742).
 
+#### Localise the character count's counter message
+
+You can now translate the text shown by the [character count](https://design-system.service.gov.uk/components/character-count/) component to inform users of:
+
+- when they reached the maximum number of characters or words
```
Comment (claireashworth) [inline CHANGELOG.md:45]:
```suggestion
- when they have reached the maximum number of characters or words
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
...
+- the number of characters or words over or under that maximum
```
Comment (claireashworth) [inline CHANGELOG.md:46]:
```suggestion
- the number of characters or words over or under the allowed maximum
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
...
+
+The Nunjucks macro accepts new options to let you customise each message:
```
Comment (claireashworth) [inline CHANGELOG.md:48]:
```suggestion
The Nunjucks macro accepts customisation for each message:
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
...
+
+- You can use `charactersAtLimitText` or `wordsAtLimitText` to provide the text to show when users reached the limit.
```
Comment (claireashworth) [inline CHANGELOG.md:50]:
```suggestion
- Use `charactersAtLimitText` or `wordsAtLimitText` to provide the text that shows when users have reached the limit.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
...
+- When users are under or over the limit, the component will pluralise the message according to the number of characters or words remaining, and the configured locale. You can pass `charactersUnderLimitText` or `wordsUnderLimitText` (when under the limit) and `charactersOverLimitText` or `wordsOverLimitText` (when over the limit) an object with the different plural forms expected by your locale. You'll find guidance about [the plural forms in our documentation about localising GOV.UK Frontend](https://design-system.service.gov.uk/get-started/localisation/#pluralisation). The component will replace `%{count}` with the number of characters over or under the limit.
```
Comment (claireashworth) [inline CHANGELOG.md:51]:
```suggestion
- When users are under or over the limit, the component will pluralise the message according to the configured locale and the number of characters or words remaining. When under the limit you can pass `charactersUnderLimitText` or `wordsUnderLimitText` and when over the limit you can pass `charactersOverLimitText` or `wordsOverLimitText`  to an object with the different plural forms expected by your locale. You'll find guidance about [the plural forms in our documentation about localising GOV.UK Frontend](https://design-system.service.gov.uk/get-started/localisation/#pluralisation). The component will replace `%{count}` with the number of characters over or under the limit.
```

Comment (claireashworth) [reply CHANGELOG.md:51]: I don't think we're putting the localisation info in the design system. I think the URL needs to be to the frontend docs (new page).

File: CHANGELOG.md [hunk 6]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
...
+
+If you're not using Nunjucks macro, you can use data-* attributes to provide these translations. Any HTML appearing within the attributes must have quotation marks and brackets converted into their [HTML entity equivalents](https://developer.mozilla.org/en-US/docs/Glossary/Entity#reserved_characters).
+
+- `data-i18n.characters-at-limit` or `data-i18n.words-at-limit` for when users are at the limit
+- `data-i18n.characters-under-limit.{other,many,few,two,one,zero}` or `data-i18n.words-under-limit.{other,many,few,two,one,zero}`, one suffix per plural form required by your locale, for when users are under the limit
+- `data-i18n.characters-over-limit.{other,many,few,two,one,zero}` or `data-i18n.words-over-limit.{other,many,few,two,one,zero}`, one suffix per plural form required by your locale, for when users are under the limit
+
+You can also provide these messages using a JavaScript configuration object when creating an instance of the component or initialising all components.
+See [our guidance on localising GOV.UK Frontend](https://design-system.service.gov.uk/get-started/localisation/) for how to do this.
```
Comment (claireashworth) [inline CHANGELOG.md:60]: we'll probably need to change this URL as well.

File: CHANGELOG.md [hunk 7]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
...
+
+- You can use `charactersAtLimitText` or `wordsAtLimitText` to provide the text to show when users reached the limit.
+- When users are under or over the limit, the component will pluralise the message according to the number of characters or words remaining, and the configured locale. You can pass `charactersUnderLimitText` or `wordsUnderLimitText` (when under the limit) and `charactersOverLimitText` or `wordsOverLimitText` (when over the limit) an object with the different plural forms expected by your locale. You'll find guidance about [the plural forms in our documentation about localising GOV.UK Frontend](https://design-system.service.gov.uk/get-started/localisation/#pluralisation). The component will replace `%{count}` with the number of characters over or under the limit.
+
+If you're not using Nunjucks macro, you can use data-* attributes to provide these translations. Any HTML appearing within the attributes must have quotation marks and brackets converted into their [HTML entity equivalents](https://developer.mozilla.org/en-US/docs/Glossary/Entity#reserved_characters).
+
+- `data-i18n.characters-at-limit` or `data-i18n.words-at-limit` for when users are at the limit
```
Comment (claireashworth) [inline CHANGELOG.md:55]:
```suggestion
- Use `data-i18n.characters-at-limit` or `data-i18n.words-at-limit` for when users are at the limit
```

File: CHANGELOG.md (hunk 3 above)
Comment (romaricpascal) [reply CHANGELOG.md:48]: My intent was to highlight that people would be able to pass new options to their Nunjucks macro, and this was a bit lost there. Would something like the following work?

```suggestion
The Nunjucks macro accepts new options to customise each message:
```

File: CHANGELOG.md (hunk 5 above)
Comment (romaricpascal) [reply CHANGELOG.md:51]: True, we'll be doing a PR to replace the links in the changelog when we'll have decided on URLs, as there are some in the accordion section as well (maybe the configuration options too). 

Regarding the content, I think the meaning changed a bit with your update. It makes it sound like users have to take an action "When under the limit" and "when over the limit". Actually, it's the component that'll make the decision to use the `...UnderLimitText` or `...OverLimitText` depending of whether the value entered is under or over the limit. Users only need to pass the keys for their message. 

In hindsight, that list item might be trying to group too many things by going for both scenarios at once. Maybe it'll work better if it's split up a bit?

```suggestion
- Use `charactersUnderLimitText` or `wordsUnderLimitText` to provide the text that shows when users are under the limit. The component will pluralise the message according to the configured locale and the number of characters or words remaining.
- Use `charactersOverLimitText` or `wordsOverLimitText` to provide the text that shows when users are over the limit. The component will pluralise the message according to the configured locale and the number of characters or words remaining.  

You'll find guidance about [the plural forms in our documentation about localising GOV.UK Frontend](https://design-system.service.gov.uk/get-started/localisation/#pluralisation). The component will replace `%{count}` with the number of characters over or under the limit.
```

That makes it a little repetitive and very close to what will be in the Nunjucks option table, but that may not be a bad thing? I'm open to other ideas

File: CHANGELOG.md [hunk 8]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
...
+- `data-i18n.characters-under-limit.{other,many,few,two,one,zero}` or `data-i18n.words-under-limit.{other,many,few,two,one,zero}`, one suffix per plural form required by your locale, for when users are under the limit
+- `data-i18n.characters-over-limit.{other,many,few,two,one,zero}` or `data-i18n.words-over-limit.{other,many,few,two,one,zero}`, one suffix per plural form required by your locale, for when users are under the limit
```
[the remark this answers is not in the document]
Comment (romaricpascal) [reply CHANGELOG.md:57]: Same update as for the previous item, if it'd work 😄 

```suggestion
- Use `data-i18n.characters-over-limit.{other,many,few,two,one,zero}` or `data-i18n.words-over-limit.{other,many,few,two,one,zero}`, one suffix per plural form required by your locale, for the message shown when users are under the limit
```

**Comment by romaricpascal** [comment]
Thanks for the review @claireashworth 😄  I've applied some of the suggestions, but a few of them where changing the meaning a little too much. Let me know if the suggestions in my comments would do. I'm not stuck on these ideas though, so if you have better alternatives, well happy to go for something else 😄

File: CHANGELOG.md (hunk 3 above)
Comment (claireashworth) [reply CHANGELOG.md:48]: Ah I see. I've edited my suggestion so the only change is from passive to active voice. Does that work?

File: CHANGELOG.md (hunk 5 above)
Comment (claireashworth) [reply CHANGELOG.md:51]: Yes, this is clearer. 👍🏻

File: CHANGELOG.md [hunk 9]
```diff
@@ -38,20 +38,45 @@ This text is announced by screen readers when the character count input is focus
...
+
+If you're not using Nunjucks macro, you can use data-* attributes to provide these translations. Any HTML appearing within the attributes must have quotation marks and brackets converted into their [HTML entity equivalents](https://developer.mozilla.org/en-US/docs/Glossary/Entity#reserved_characters).
+
+- `data-i18n.characters-at-limit` or `data-i18n.words-at-limit` for when users are at the limit
+- `data-i18n.characters-under-limit.{other,many,few,two,one,zero}` or `data-i18n.words-under-limit.{other,many,few,two,one,zero}`, one suffix per plural form required by your locale, for when users are under the limit
```
[the remark this answers is not in the document]
Comment (claireashworth) [reply CHANGELOG.md:56]: For all of the sections where the changes make it sound like the user is causing the change how about amending it to:
"To show text that informs the end user they are under the character or word limit, use...."
"To show text that informs the end user they are over the character or word limit, use..."
?

File: CHANGELOG.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (claireashworth) [reply CHANGELOG.md:57]: I've edited my suggestion as per my previous comment. What do you think?

File: CHANGELOG.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (romaricpascal) [reply CHANGELOG.md:56]: I think that still makes it sound like it's the user taking an action, while those attributes are just configuration that need setting once before initialising the component. How about a variation of your last proposition?

"To configure the text that informs the end user they are under the character or word limit, use..."
"To configure the text that informs the end user they are over the character or word limit, use..."

[the remark this answers is not in the document]
Comment (romaricpascal) [reply CHANGELOG.md:59]:
```suggestion
- To configure the text that informs the end user they are under the character or word limit, use `data-i18n.characters-under-limit.{other,many,few,two,one,zero}` or `data-i18n.words-under-limit.{other,many,few,two,one,zero}`, with one suffix per plural form required by your locale
```

File: CHANGELOG.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (romaricpascal) [reply CHANGELOG.md:60]:
```suggestion
- To configure the text that informs the end user they are over the character or word limit, use `data-i18n.characters-over-limit.{other,many,few,two,one,zero}` or `data-i18n.words-over-limit.{other,many,few,two,one,zero}`, with one suffix per plural form required by your locale
```

**Review by claireashworth — APPROVED** [review]

**Pull request merged**